src/viewport/viewport-matcher-var.directive.ts

```diff
@@ -0,0 +1,87 @@
+import {
+	OnInit,
+	OnDestroy,
+	Directive,
+	Input,
+	ChangeDetectorRef,
+	TemplateRef,
+	ViewContainerRef,
+} from "@angular/core";
+import { combineLatest, ReplaySubject, Subject } from "rxjs";
+import { tap, map, takeUntil } from "rxjs/operators";
+
+import { ViewportService } from "./viewport.service";
+import {
+	isViewportSizeMatcherExpression,
+	isViewportSizeMatcherTupleExpression,
+	isViewportConditionMatch
+} from "./viewport.util";
+import { ViewportMatchConditions } from "./viewport.model";
+
+const NAME_CAMEL = "ssvViewportMatcherVar";
+
+export class SsvViewportMatcherVarContext {
+
+	constructor(
+		public $implicit = false,
+	) { }
+
+}
+
+@Directive({
+	selector: `[${NAME_CAMEL}]`,
+	exportAs: NAME_CAMEL,
```

Comment:
Note that `exportAs` does not work with sugared syntax, can still use it if no sugar though so can keep it

Reply:
infect i had my doubities but saw no errors so I thought it was fine :/

Reply:
actually i dont think you can with a structural, and it doesn't offer anything so I will remove anyway